kymatio/scattering1d/backend/torch_skcuda_backend.py

```diff
@@ -62,29 +62,29 @@ def __call__(self, x):
         if not x.is_cuda and self.backend=='skcuda':
             raise TypeError('Use the torch backend (without skcuda) for CPU tensors.')
 
-        out = x.new(x.shape)
+        out = x.new(x.shape[:-1] + (1,))
 
         if not x.is_contiguous():
             raise RuntimeError('Input should be contiguous.')
 
         if not _is_complex(x):
-            raise TypeError('The input and outputs should be complex.')
+            raise TypeError('The input should be complex.')
 
         kernel = """
         extern "C"
-        __global__ void abs_complex_value(const ${dtype} * x, ${dtype}2 * z, int n)
+        __global__ void abs_complex_value(const ${dtype} * x, ${dtype} * z, int n)
         {
             int i = blockIdx.x * blockDim.x + threadIdx.x;
         if (i >= n)
             return;
-        z[i] = make_${dtype}2(normf(2, x + 2*i), 0);
+        z[i] = normf(2, x + 2*i);
```

Comment:
Why is this casting operation gone?

Reply:
I will be honest and say I don't know. It seemed like `make_${dtype}2(normf(2, x + 2*i), 0);	` is saying to make the last dimension of size 2, and have the imaginary part be 0. I can't really read cuda code tho. I remember removing the 0 and it said something about being unable to convert from float to float2 or something like that.

Reply:
> It seemed like make_${dtype}2(normf(2, x + 2*i), 0); is saying to make the last dimension of size 2, and have the imaginary part be 0.

I'm fairly certain that's it.

Reply:
Can we, before merging this, get me a clear view of what's going on here?

This function is called `abs_complex_value`. Is it still being applied to complex values? But now its output is an array of reals?

Reply:
Yes, its being applied to complex values, then going into an array of reals. should i change the name to `abs`?